File: solutions/0153-find-minimum-in-rotated-sorted-array/solution.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        
    int l=0,r=nums.size()-1;
    if(nums[l]<nums[r]) return nums[l];
    int min = nums[0];

    while (l <= r) {
        if (nums[l] < min)
            min = nums[l];
        if (nums[r] < min)
            min = nums[r];

        l++;
        r--;
    }

    return min;
    }
};

File: solutions/0153-find-minimum-in-rotated-sorted-array/solution.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l,r=0,len(nums)-1
        if nums[l]<nums[r]: return nums[l];
        m=nums[0]
        while(l<=r):
            m=min(m,nums[l],nums[r])
            l+=1
            r-=1
        return m
